src/operators/semesterOperations.ts

```diff
@@ -0,0 +1,48 @@
+import Semester, { SemesterInterface } from '../models/semester';
+
+export const create = async (semester: SemesterInterface) => {
+	try {
+		await Semester.create(semester);
+	} catch (err) {
+		throw 'ERROR: semester not created';
+	}
+};
+
+// NOTE: any reference to key is referring to the hash key, or the `id` parameter in all the models.
+export const read = async (key: string) => {
+	// add some kind of security check here?
+	const semester = await Semester.get(key);
+	if (semester) {
+		return semester;
+	} else {
+		throw 'ERROR: could not read semester';
+	}
+};
+
+export const update = async (data: Partial<SemesterInterface>) => {
+	let updated = false;
+	if (typeof data.id == 'string') {
+		const key: string = data.id;
+		const semester = await Semester.get(key);
+		if (semester && (await Semester.update(data))) {
+			updated = true;
+		}
+	}
+	if (!updated) {
+		throw 'ERROR: semester not updated';
+	}
```

Comment:
Similar to above we can probably simplify the logic

```ts
export const update = async (data: Partial<SemesterInterface>) => {
    if(data.id) { // This check will suffice for seeing if id is defined on our partial object
        const key = data.id; // This should automatically know that id is a string... if that isn't the case, lmk because it is likely an issue elsewhere
        const semester = await Semester.get(key);
        // Should probably do a check that the semester is owned by the caller!
        // Again, we know semester has resolved because of the throwing behaviour
        await Semester.update(data); 
        // Wondering if we should return Semester.update(data);
    }
}

Reply:
Per your discovery you may have to do a couple throw checks (if the semester isn't found, and if the owner is not the person making the request)